################################################################################
SHORT DESCRIPTION: 
################################################################################
Modify libxl to pass ipxe ROM for seabios.

################################################################################
LONG DESCRIPTION: 
################################################################################
Upstream xen only supports passing the ipxe file to hvmloader if we're using
rombios. Because we use seabios, we need to disable the check for rombios
so we can use --with-system-ipxe, and enable the codepath if the bios we're
using is seabios in libxl_dom.c

commit bcf77ce51004 "configure: modify default of building rombios"
changed configure so ipxe no longer depends on rombios to build.

--- a/tools/libs/light/libxl_dom.c
+++ b/tools/libs/light/libxl_dom.c
@@ -981,7 +981,8 @@ static int libxl__domain_firmware(libxl_
     }
 
     if (info->type == LIBXL_DOMAIN_TYPE_HVM &&
-        info->u.hvm.bios == LIBXL_BIOS_TYPE_ROMBIOS &&
+        (info->u.hvm.bios == LIBXL_BIOS_TYPE_ROMBIOS ||
+         info->u.hvm.bios == LIBXL_BIOS_TYPE_SEABIOS) &&
         libxl__ipxe_path()) {
         const char *fp = libxl__ipxe_path();
         rc = xc_dom_module_file(dom, fp, "ipxe");
